eclipse-theia/theia
# Fix unhandled promise rejection during git operations

#### What it does
This fixes an issue we've seen happening on KSC when there's a conflict while pulling git changes from a remote.

It seems that when exception is thrown from the dugite-git library, an unhandled promise rejection is caught by the 'unhandledRejection' handler which rethrows the error and ends up in the 'uncaughtException' handler, which caused the container to crash on kubernetes. On a local docker compose setup, the container doesn't crash.

#### How to test
* Clone a repository in theia
* In another clone of the same repository, add some extra commits
* Back in theia, change one of the files from previous step
* Pull the changes, it should give you an error message of the conflict
* In the logs, the following should not be visible:

```
2023-04-19T13:35:48.732Z root ERROR Uncaught Exception:  GitError: Your local changes to the following files would be overwritten
```

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
